lib/spack/spack/cmd/checkbinaries.py

```diff
@@ -0,0 +1,247 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from __future__ import print_function
+
+from six.moves.urllib.request import urlopen
+from six.moves.urllib.error import URLError
+
+import json
+import os
+import re
+import yaml
+
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.config
+from spack.error import SpackError
+from spack.spec import Spec
+from spack.util.executable import which
+
+
+description = "check if binaries need to be rebuilt"
+section = "developer"
+level = "long"
+
+
+_buildcache_path = os.path.join('build_cache')
```

Comment:
I don't think this needs an `os.path.join`.  Shouldn't this location be taken from `binary_distribution`?  Oh... I just looked at that.  It's not defined as a variable there, it's just a string written in lots of places... ugh.  Can you make it into a variable in `binary_distribution`?

Reply:
Yes, I was thinking the same thing here.  I'll do it in my next pass.